sdmetrics/reports/single_table/_properties/column_pair_trends.py

```diff
@@ -0,0 +1,455 @@
+            metadata (dict):
+                The metadata of the table
+        """
+        sdtype_col_1 = metadata['columns'][column_name_1]['sdtype']
+        sdtype_col_2 = metadata['columns'][column_name_2]['sdtype']
+        if self._sdtype_to_shape[sdtype_col_1] != self._sdtype_to_shape[sdtype_col_2]:
+
+            if self._sdtype_to_shape[sdtype_col_1] == 'continuous':
+                col_1 = discrete_data[column_name_1]
+                col_2 = data[column_name_2]
+            else:
+                col_1 = data[column_name_1]
+                col_2 = discrete_data[column_name_2]
+
+            return pd.concat([col_1, col_2], axis=1)
+        else:
+            return data[[column_name_1, column_name_2]]
+
+    def _preprocessing_failed(self, column_name_1, column_name_2, sdtype_col_1, sdtype_col_2):
+        """Check if a processing of one of the columns has failed.
+
+        Args:
+            column_name_1 (str):
+                The name of the first column
+            column_name_2 (str):
+                The name of the second column
+            sdtype_col_1 (str):
+                The sdtype of the first column
+            sdtype_col_2 (str):
+                The sdtype of the second column
+        """
+        error = None
+        if column_name_1 in self._columns_datetime_conversion_failed.keys():
+            error = self._columns_datetime_conversion_failed[column_name_1]
+
+        elif column_name_2 in self._columns_datetime_conversion_failed.keys():
+            error = self._columns_datetime_conversion_failed[column_name_2]
+
+        elif self._sdtype_to_shape[sdtype_col_1] != self._sdtype_to_shape[sdtype_col_2]:
+            if column_name_1 in self._columns_discretization_failed.keys():
+                error = self._columns_discretization_failed[column_name_1]
+            elif column_name_2 in self._columns_discretization_failed.keys():
+                error = self._columns_discretization_failed[column_name_2]
+
+        return error
+
+    def _get_score_breakdown(
+            self, metric, col_name_1, col_name_2, real_data, discrete_real,
+            synthetic_data, discrete_synthetic, metadata):
+        """Get the score breakdown for the property.
+
+        Args:
+            metric (object):
+                The metric to use
+            col_name_1 (str):
+                The name of the first column
+            col_name_2 (str):
+                The name of the second column
+            real_data (pandas.DataFrame):
+                The real data
+            synthetic_data (pandas.DataFrame):
+                The synthetic data
+            metadata (dict):
+                The metadata of the table
+        """
+        columns_real = self._get_columns_data(
+            col_name_1, col_name_2, real_data, discrete_real, metadata
+        )
+        columns_synthetic = self._get_columns_data(
+            col_name_1, col_name_2, synthetic_data, discrete_synthetic, metadata
+        )
```

Comment:
I think you can do the logic for these methods and getting the metric itself all in here since they seem to share similar conditional statements